### ISSUE TITLE
Serialize and deserialize a neural network without using a file

### ISSUE THREAD

**Issue by arrufat** [davisking/dlib#498]
Hi!
I'm training a neural network and I am using a validation test set to guide the training every `train_one_step`.
The network used on the validation is the same one as the one in the training, except for the `bn_conv`, which are `affine`.
So I'd like to serialize the training network into the testing network without accessing to a temporary file.
Is that possible?
Thanks in advance :)

**Comment by davisking** [comment]
Just assign the networks with =

**Comment by arrufat** [comment]
Thank you very much... So simple in the end...
